### ISSUE TITLE
locateOnScreen without saving temporary image file

### ISSUE THREAD

**Issue by BenRoe** [asweigart/pyautogui#379]
Hello,
is the normal behaviour of the `locateOnScreen()` function that it saves a temporary image file to the project folder?
```python
img_region = pyautogui.locateOnScreen(
            imgPath(img_name), region=region, grayscale=False, confidence=precision)
```

I wrote a bot which is running on a ssd and i am worried, that constantly writing the temporary image to the ssd will damage it?

If i use `image = pyautogui.screenshot()` no temporary images are written to the project folder.

**Comment by Exaxxion** [comment]
I've been dabbling with this library for desktop automation and I have the same concern as above that was never addressed.

If I monitor the screen in a loop, it rapidly spams and deletes several MB file writes to disk ".screenshot(timestamp).png".

This seems terrible for disk life and I'm wondering if there's a way to monitor regions of the screen entirely in memory to avoid unnecessary wear on my drive.

**Comment by purificant** [comment]
On Linux, you need to install `gnome-screenshot` to process images in memory with `pillow` and avoid creating temporary files on disk. See [this code](https://github.com/asweigart/pyscreeze/blob/master/pyscreeze/__init__.py#L596) for details.

**Comment by Exaxxion** [comment]
That's interesting to hear. Unfortunately I do not use Gnome DE, so I do not have that available. I use a DE called MATE, which was forked from Gnome 2, which has a program called `mate-screenshot`.

Looking further in that code block, it seems that it first tries to use `gnome-screenshot` as a compatibility for Wayland (which I am also not using, I am using x11). It seems to fall back to `scrot` which is installed on my system, and always outputs to a temporary file.

Using the python interpreter on my system, I am able to import `ImageGrab` from Pillow 9.2.0, invoke `im = ImageGrab.grab()` and evaluating `im` displays `<PIL.Image.Image image mode=RGB size=... at ...>`.

That pyscreeze fails to attempt this appears to be its own shortcoming.

**Comment by purificant** [comment]
You are right @Exaxxion , there's probably a better way to handle [this](https://github.com/asweigart/pyscreeze/blob/master/pyscreeze/__init__.py#L595) in pyscreeze. There's probably a way to clean up this code because py3.7 is end of life now, and can also use [tempfile](https://docs.python.org/3/library/tempfile.html#module-tempfile) module with a [context manager](https://docs.python.org/3/library/tempfile.html#examples) in handle temporary files in /tmp, which on Linux is sometimes mounted in RAM therefore eliminating any disk I/O.

As a workaround for your specific use case, can symlink `mate-screenshot` to `gnome-screenshot` which will then cause the code path in pyscreeze to use `Pillow` without creating a temporary file on disk.
```bash
sudo ln -s /usr/bin/mate-screenshot /usr/bin/gnome-screenshot
```

**Comment by Exaxxion** [comment]
Creating a symlink seemed to confuse pyscreeze, as it would expect a file present in `/tmp` but it was 0 bytes and the program would crash.

I modified my installed copy of pyscreeze to disregard whether `gnome-screenshot` is installed so long as Pillow 9.2.0+ is present, and disabled the flag that made it throw exceptions, because when it routed into the opencv path, it would throw exceptions for no matches instead of returning `None` as before.

Now it works much better and it is not generating temporary files.
